map.js

```diff
@@ -147,29 +130,34 @@ function drawMapOverlay(){
             icon.recalculateBoundingBox();
         }
```

Comment:
(this comment is for the beginning of drawMapOverlay)
This could be a method on the "overlay" object. I didn't move it there because the overlay actually doesn't really do anything by itself. its just a collection of map icons, which *do* do things on their own.
ex: overlay.contains() is pointless, since the overlay spans the entire map. so of COURSE any click will hit the overlay.

I will explain my decisions for which properties to keep in MapIcon at the mapicon code.